sds_data_manager/constructs/compute_resources.py

```diff
@@ -204,3 +204,101 @@ def __init__(
             },
             tags={"Purpose": "Batch Processing"},
         )
+
+
+class IalirtEC2Resources(Construct):
+    """EC2 compute environment."""
+
+    def __init__(
+        self,
+        scope: Construct,
+        construct_id: str,
+        vpc: ec2.Vpc,
+        ecr_policy: iam.PolicyStatement,
+        repo: ecr.Repository,
+        instance_type: str = "t3.micro",
+    ):
+        """Constructor
+
+        Parameters
+        ----------
+        scope : Construct
+            Parent construct.
+        construct_id : str
+            A unique string identifier for this construct.
+        vpc : ec2.Vpc
+            VPC in which to create compute instances.
+        ecr_policy : iam.PolicyStatement
+            ECR policy statement.
+        repo : ecr.Repository
+            Container repo.
+        instance_type : str, Optional
+            Type of EC2 instance to launch.
+        """
+        super().__init__(scope, construct_id)
+
+        # Define user data script
+        # - Updates the instance
+        # - Installs Docker
+        # - Starts the Docker
+        # - Logs into AWS ECR
+        # - Pulls the Docker image
+        # - Runs the Docker container
+        user_data = ec2.UserData.for_linux()
+        user_data.add_commands(
+            "yum update -y",
+            "amazon-linux-extras install docker -y",
+            "systemctl start docker",
+            "systemctl enable docker",
+            "$(aws ecr get-login --no-include-email --region us-west-2 | bash)",
+            f"docker pull {repo.repository_uri}:latest",
+            f"docker run --rm -d -p 8080:8080 {repo.repository_uri}:latest",
+        )
+
+        # Security Group for the EC2 Instance
+        security_group = ec2.SecurityGroup(
+            self,
+            "IalirtEC2SecurityGroup",
+            vpc=vpc,
+            description="Security group for Ialirt EC2 instance",
+        )
+
+        # Allow ingress to LASP IP address range and specific port
+        security_group.add_ingress_rule(
+            ec2.Peer.ipv4("128.138.131.0/24"),
+            ec2.Port.tcp(8080),
+            "Allow inbound traffic on TCP port 8080",
+        )
+
+        # Create an IAM role for the EC2 instance
+        # - Read-only access to AWS ECR
+        # - Basic instance management via AWS Systems Manager
+        ec2_role = iam.Role(
+            self,
+            "IalirtEC2Role",
+            assumed_by=iam.ServicePrincipal("ec2.amazonaws.com"),
+            managed_policies=[
+                iam.ManagedPolicy.from_aws_managed_policy_name(
+                    "AmazonEC2ContainerRegistryReadOnly"
+                ),
+                iam.ManagedPolicy.from_aws_managed_policy_name(
+                    "AmazonSSMManagedInstanceCore"
+                ),
+            ],
+        )
+        ec2_role.add_to_policy(ecr_policy)
```

Comment:
Do you need the `ecr_policy` here too? You're adding the ECR ReadOnly above, so I think this might be expanding the permissions.